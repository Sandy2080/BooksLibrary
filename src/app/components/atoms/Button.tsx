import React from "react";

export enum Size {
    sm = "btn-sm", lg = "btn-lg"  
}
export const Info = (children: any, size: Size, style: string) => {
    return(<button className={`btn btn-info ${size} ${style}`}>{children}</button>)
}
export const Success = (children: string, size: Size) => {
  return <button className={`btn btn-success ${size}`}>{children}</button>;
};
export const Warning = (children: string, size: Size) => {
  return <button className={`btn btn-warning ${size}`}>{children}</button>;
};
   
    

