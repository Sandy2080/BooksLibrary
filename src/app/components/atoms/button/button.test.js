import React from 'react';
import Button, { ButtonTheme, ButtonSize } from './index';
import renderer from 'react-test-renderer';

describe('Button Primary', () => {
  it('renders correctly when button is PRIMARY  ', () => {
    const tree = renderer.create(<Button.PRIMARY />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Button Rounded', () => {
  it('renders correctly when button is ROUNDED and LARGE', () => {
    const tree = renderer
      .create(<Button.PRIMARY size={ButtonSize.LARGE} theme={ButtonTheme.ROUNDED} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
