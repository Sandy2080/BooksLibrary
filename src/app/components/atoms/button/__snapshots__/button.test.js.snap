// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Primary renders correctly when button is PRIMARY   1`] = `
<button
  className="sc-AykKC ifcwsQ btn btn-primary"
  disabled={false}
/>
`;

exports[`Button Rounded renders correctly when button is ROUNDED and LARGE 1`] = `
<button
  className="sc-AykKC nBkZa btn btn-primary btn-lg"
  disabled={false}
  size="btn-lg"
/>
`;
